nathanwkwong/node-ts-banking
# Project configuration and express server setup

## What does this PR do?

init project

## Related Projects/Specs/Tasks (if any)?

- add a docker file for development and build stages
-  set up db connection
- set up TypeORM migration config and command to sync with db
- add a simple express server
- add test config for unit, integration, and e2e tests
- setup prettier, eslint rules and husky pre-commit hook

## How should this be manually tested?

1. setup a .env file at the root following .env. example
2. run the `docker-compose up` for a local Postgres container
3. run `npm urn dev` for a local express server to see whether it can connect to the Postgres db and see the migrations work 
4. test prettier and eslint checking during commit by adding a line of code that violates the rules and verifying the line of code can't be committed
6. test build stages with docker by running the Docker file and will dockerize and build the express server inside a docker image

## Any background context you want to provide?(Optional)

(Add any information that will further explain or provide extra context on what drove you to create this PR)

## What are the relevant Notion URLs?

- prettier, eslint: https://www.notion.so/Secure-Banking-Application-11c0975b4ef3805bb06dc4be20f0cffa?p=1240975b4ef380eeb43aea65d1283904&pm=s
- typeorm and : https://www.notion.so/Secure-Banking-Application-11c0975b4ef3805bb06dc4be20f0cffa?p=e170ac23a1b448a38d2eb3c7fc148dda&pm=s
- express and typescript setting https://www.notion.so/Secure-Banking-Application-11c0975b4ef3805bb06dc4be20f0cffa?p=b696d334c90c4b5682e057b1a0b353d6&pm=s
- dockerize: https://www.notion.so/Secure-Banking-Application-11c0975b4ef3805bb06dc4be20f0cffa?p=1240975b4ef3802dbd76fc983a5fa9a1&pm=s
- jest test: https://www.notion.so/Secure-Banking-Application-11c0975b4ef3805bb06dc4be20f0cffa?p=1240975b4ef3803d83c9d2ece6368e4a&pm=s
